CMakeLists.txt

```diff
@@ -224,19 +226,26 @@ if (NN_ENABLE_NANOCAT)
     target_link_libraries (nanocat ${PROJECT_NAME})
 endif ()
 
-if (NN_ENABLE_DOC)
-    find_program (ASCIIDOCTOR_EXE asciidoctor)
-    if (NOT ASCIIDOCTOR_EXE)
-        message (WARNING "Could not find asciidoctor: skipping docs")
-        set (NN_ENABLE_DOC OFF)
-    else ()
+# Build the documentation
+string (TOUPPER "${NN_ENABLE_DOC}" NN_ENABLE_DOC)
+if (NN_ENABLE_DOC STREQUAL "OFF")
+    set (NN_BUILD_DOC OFF)
+elseif (NN_ENABLE_DOC STREQUAL "DEFAULT" OR NN_ENABLE_DOC STREQUAL "ON")
+    find_program (NN_ASCIIDOCTOR_INSTALLED asciidoctor)
+    if (NN_ASCIIDOCTOR_INSTALLED)
         message (STATUS "Using asciidoctor at ${ASCIIDOCTOR_EXE}")
+        set (NN_BUILD_DOC ON)
+    elseif (NN_ENABLE_DOC STREQUAL "DEFAULT")
+        message (WARNING "Could not find asciidoctor; skipping docs.")
+        set (NN_BUILD_DOC OFF)
+    else ()
+        message (FATAL_ERROR "asciidoctor is not installed. (hint: use -DNN_ENABLE_DOC=DEFAULT to suppress this error)")
     endif ()
+else ()
+    message (FATAL_ERROR "Unknown option: NN_ENABLE_DOC=${NN_ENABLE_DOC} (hint: use ON|OFF|DEFAULT)")
 endif ()
```

Comment:
Ah, I see, you have explicit checks here. Are ON/OFF sufficient to emulate what CMake does today?